### DISCUSSION TITLE
Error while updating MOOSE

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#26778] [Q&A General]
My moose was working fine until I tried to update it today. Below is the error I get

```
python-stack) [emmanuel324@tinkercliffs1 belson]$ cd ~/projects/moose
(python-stack) [emmanuel324@tinkercliffs1 moose]$ 
(python-stack) [emmanuel324@tinkercliffs1 moose]$ git fetch origin
(python-stack) [emmanuel324@tinkercliffs1 moose]$ 
(python-stack) [emmanuel324@tinkercliffs1 moose]$ git rebase origin/master
Cannot rebase: You have unstaged changes.
Please commit or stash them.
(python-stack) [emmanuel324@tinkercliffs1 moose]$ cd ~/projects/belson
(python-stack) [emmanuel324@tinkercliffs1 belson]$ make -j 6
Not compiling MOOSE with NEML2 because NEML2_DIR is not a valid NEML2 checkout and/or libtorch is not enabled.
***ERROR***
WASP does not seem to be available.
Make sure to either run scripts/update_and_rebuild_wasp.sh in your MOOSE directory,
or set WASP_DIR to a valid WASP install
make: *** [wasp_submodule_status] Error 1
(python-stack) [emmanuel324@tinkercliffs1 belson]$ ./run_tests -j 6
Building and linking hit...
/home/emmanuel324/projects/moose/framework/contrib/hit/parse.cc: In member function ‘hit::Field::Kind hit::Field::kind()’:
/home/emmanuel324/projects/moose/framework/contrib/hit/parse.cc:684:34: error: ‘class wasp::HITNodeView’ has no member named ‘token_type’; did you mean 
‘set_type’?
  684 |     auto token_type = value_node.token_type();
      |                                  ^~~~~~~~~~
      |                                  set_type
make: *** [hit] Error 1
Traceback (most recent call last):
  File "/home/emmanuel324/projects/moose/python/pyhit/__init__.py", line 13, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/emmanuel324/projects/moose/python/pyhit/__init__.py", line 18, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/emmanuel324/projects/belson/./run_tests", line 11, in <module>
    from TestHarness import TestHarness
  File "/home/emmanuel324/projects/moose/python/TestHarness/__init__.py", line 15, in <module>
    from .TestHarness import TestHarness
  File "/home/emmanuel324/projects/moose/python/TestHarness/TestHarness.py", line 20, in <module>
    from FactorySystem.Factory import Factory
  File "/home/emmanuel324/projects/moose/python/FactorySystem/__init__.py", line 13, in <module>
    from .Parser import Parser
  File "/home/emmanuel324/projects/moose/python/FactorySystem/Parser.py", line 13, in <module>
    import pyhit
  File "/home/emmanuel324/projects/moose/python/pyhit/__init__.py", line 22, in <module>
    import hit
ModuleNotFoundError: No module named 'hit'
(python-stack) [emmanuel324@tinkercliffs1 belson]$ 
```

**Comment by GiudGiud** [comment]
Hello

A few things here:
- did you update the moose packages on conda? (`conda update moose-libmesh`)
- your rebase failed. Can you commit all your changes then try again?

**Reply by Emmanuel-324** [reply]
This is what i used. And how do I commit my changes?

cd ~/projects/moose
git fetch origin
git rebase origin/master

**Reply by GiudGiud** [reply]
after `git rebase origin/master` you see it fail

do this:
`git commit -am "My changes"`

**Reply by Emmanuel-324** [reply]
Should I try updating the moose again.

**Reply by GiudGiud** [reply]
yeah once you have commited you can update libmesh first then rebase your branch of moose

**Reply by Emmanuel-324** [reply]
the command i replace with "My changes", is it the directory containing the changes?

**Reply by Emmanuel-324** [reply]
All the changes I made are in my own application

This is the result I am getting trying to commit

git: 'clang-format' is not a git command. See 'git --help'.
Your code is not properly formatted.
Run git clang-format to resolve the following issues:
(python-stack) [emmanuel324@tinkercliffs1 step01_diffusion]$ git commit -am "problems"
git: 'clang-format' is not a git command. See 'git --help'.
Your code is not properly formatted.
Run git clang-format to resolve the following issues:
(python-stack) [emmanuel324@tinkercliffs1 step01_diffusion]$

**Comment by Emmanuel-324** [comment]
All the changes I made are in my own application

This is the result I am getting trying to commit

git: 'clang-format' is not a git command. See 'git --help'.
Your code is not properly formatted.
Run git clang-format to resolve the following issues:
(python-stack) [emmanuel324@tinkercliffs1 step01_diffusion]$ git commit -am "problems"
git: 'clang-format' is not a git command. See 'git --help'.
Your code is not properly formatted.
Run git clang-format to resolve the following issues:
(python-stack) [emmanuel324@tinkercliffs1 step01_diffusion]$

**Reply by GiudGiud** [reply]
This looks like a problem with your git hooks. 
Maybe you should activate the moose conda environment?

**Reply by Emmanuel-324** [reply]
How do I activate the moose conda environment?

**Reply by GiudGiud** [reply]
`conda activate moose`

**Reply by Emmanuel-324** [reply]
(python-stack) [emmanuel324@tinkercliffs1 ~]$ conda activate moose
EnvironmentNameNotFound: Could not find conda environment: moose
You can list all discoverable environments with `conda info --envs`.

**Reply by GiudGiud** [reply]
how did you install moose? which instructions did you follow?

**Reply by Emmanuel-324** [reply]
I actually followed instructions from someone from Moose because I was finding it difficult to get a current cmake to load my environment. I want to see if I can find the thread
